core/trino-main/src/main/java/io/trino/sql/planner/plan/TableDeleteNode.java

```diff
@@ -76,6 +77,7 @@ public Symbol getOutput()
     @Override
     public PlanNode replaceChildren(List<PlanNode> newChildren)
     {
-        return new TableDeleteNode(getId(), target, output);
+        checkArgument(newChildren.isEmpty(), "newChildren should be empty");
```

Comment:
let's move "Stop ignoring children in TableDeleteNode.replaceChildren" to a separate PR for faster merge.